### DISCUSSION TITLE
Misalignment errors in PyAT

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#295] [Q&A] [answered]
Dear all,

I want to insert a misalignment error to the quadrupole using PyAT
These are the only parameters that i have :
My question is how to control the misalignment errors in PyAT ?

Quadrupole(FamName, Length, Strength=0, **keywords)

Available keywords:
PolynomB        straight multipoles
PolynomA        skew multipoles
MaxOrder        Number of desired multipoles
NumIntSteps     Number of integration steps (default: 10)
FringeQuadEntrance  0: no fringe fiels effect (default)
                    1: Lee-Whiting's thin lens limit formula
                    2: elegant-like
FringeQuadExit
fringeIntM0     Integrals for FringeQuad method 2
fringeIntP0
KickAngle       Correction deviation angles (H, V)

**Comment by lfarv** [comment]
The misalignment of a PyAT element is set by optional attributes:
- T1 and T2 for entrance and exit translations,
- R1 and R2 for entrance and exit rotation along the s axis (magnet tilt)

Though you can set these attributes manually, an easier way to do this is to use the `shift_elem` (for a single element) or `set_shift` (for a full set f elements) functions. And similarly, for tilts there are `tilt_elem` and `set_tilt`.

**Reply by elafmusa** [reply]
@lfarv Thank you, i tried it and it works, it seems that the pervious misalignment value is kept in memory. Is there a way with which i can reset the ring to the initial state ?

**Comment by lfarv** [comment, marked as answer]
The misalignment values are kept because they are part of the element. To reset, you may either use the same functions to set the values to zero, or delete the element attributes (`del element.T1` and so on).

**Reply by elafmusa** [reply]
@lfarv Thank you for the clarification.
